src/librustc_mir/transform/mod.rs

```diff
@@ -185,6 +184,41 @@ pub fn run_passes(
     body.phase = mir_phase;
 }
 
+fn mir_const_qualif(tcx: TyCtxt<'_>, def_id: DefId) -> u8 {
+    let const_kind = check_consts::ConstKind::for_item(tcx, def_id);
+
+    // No need to const-check a non-const `fn`.
+    if const_kind.is_none() {
+        return 0;
+    }
+
+    // N.B., this `borrow()` is guaranteed to be valid (i.e., the value
+    // cannot yet be stolen), because `mir_validated()`, which steals
+    // from `mir_const(), forces this query to execute before
+    // performing the steal.
+    let body = &tcx.mir_const(def_id).borrow();
+
+    if body.return_ty().references_error() {
+        tcx.sess.delay_span_bug(body.span, "mir_const_qualif: MIR had errors");
+        return check_consts::QualifSet::UNPROMOTABLE.into();
```

Comment:
Does anything break if you use `QualifSet::default()` here?

Reply:
Apparently not :smiley: